src/gui/TrackContentObjectView.cpp

```diff
@@ -568,7 +578,35 @@ void TrackContentObjectView::mousePressEvent( QMouseEvent * me )
 	setInitialOffsets();
 	if( !fixedTCOs() && me->button() == Qt::LeftButton )
 	{
-		if( me->modifiers() & Qt::ControlModifier )
+		if( m_trackView->trackContainerView()->knifeMode() )
+		{
+			SampleTCO * sTco = dynamic_cast<SampleTCO*>( m_tco );
+
+			if( me->x() < RESIZE_GRIP_WIDTH && sTco
+					&& !m_tco->getAutoResize() )
+			{
+				m_action = ResizeLeft;
+				setCursor( Qt::SizeHorCursor );
+			}
+			else if( me->x() >= width() - RESIZE_GRIP_WIDTH )
+			{
+				m_action = Resize;
+				setCursor( Qt::SizeHorCursor );
+			}
+			else if (sTco)
+			{
+				m_action = Split;
+				setCursor( m_cursorKnife );
+				sTco->setMarkerPos( knifeMarkerPos( me ) );
+				sTco->setMarkerEnabled( true );
+				update();
+			}
+			// We can't split anything except samples right now, so disable the
+			// action to avoid entering if statements we don't need to. This
+			// also saves us a few 'if(sTco)' checks
+			else { m_action = NoAction; }
```

Comment:
I updated the suggestion, I think it's better to surround the whole block with `if (sTco && !m_tco->getAutoResize())` because it might be a problem to split a resizable TCO (since the splitting itself involves resizing of clips).

Reply:
I forgot to mention this earlier, but I can't reproduce this issue. Regardless of whether I have knife mode enabled in the song editor or not I can't cut in the BB editor.